pkg/controller/s3/bucket/requestPaymentConfig.go

```diff
@@ -104,3 +89,27 @@ func (in *RequestPaymentConfigurationClient) CreateOrUpdate(ctx context.Context,
 func (*RequestPaymentConfigurationClient) Delete(_ context.Context, _ *v1beta1.Bucket) error {
 	return nil
 }
+
+// LateInitialize is responsible for initializing the resource based on the external value
+func (in *RequestPaymentConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketRequestPaymentRequest(&awss3.GetBucketRequestPaymentInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		return awsclient.Wrap(err, paymentGetFailed)
+	}
+	if len(external.Payer) == 0 {
+		return nil
+	}
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+	config := bucket.Spec.ForProvider.PayerConfiguration
+	if config == nil {
+		bucket.Spec.ForProvider.PayerConfiguration = &v1beta1.PaymentConfiguration{}
+		config = bucket.Spec.ForProvider.PayerConfiguration
```

Comment:
Already assigned in line `103` and honestly I don't really get the point of this redirection, why not just keep using `bucket.Spec.ForProvider.PayerConfiguration`?

Reply:
Yup, this is definitely an ugly pattern, I've gone through and fixed it where I've noticed it now. I am not sure exactly what motivated me to write it this way before